model/task/task.go

```diff
@@ -2493,8 +2491,14 @@ func (t *Task) Archive() error {
 		return errors.Wrap(err, "inserting archived task into old tasks")
 	}
 	err = UpdateOne(
-		bson.M{IdKey: t.Id},
 		bson.M{
+			IdKey:     t.Id,
+			StatusKey: bson.M{"$in": evergreen.TaskCompletedStatuses},
+		},
+		bson.M{
+			"$set": bson.M{
+				ArchivedKey: true,
```

Comment:
This seems to be misusing the usage of Archive. Archive is supposed to be set only on tasks that are being inserted into the old task collection (i.e. makeArchivedTask).

Reply:
Yes currently the Archived field is unused in the active tasks collection but this change is extending the use of the Archived field to be used as an indicator that the task is in a state where it is safe to reset and doing so will not result in task conflict errors.